src/multibody/liegroup/liegroup-base.hxx

```diff
@@ -400,12 +530,70 @@ namespace pinocchio {
       const Eigen::MatrixBase<ConfigL_t> & q0,
       const Eigen::MatrixBase<ConfigR_t> & q1) const
   {
-    TangentVector_t diff;
+    TangentVector_t diff(nv());
     difference(q0.derived(), q1.derived(), diff);
     return diff;
   }
 
   // ----------------- Default implementations ------------------------------ //
+  template <class Derived>
+  template <class Config_t, class Tangent_t, class JacobianIn_t, class JacobianOut_t>
+  void LieGroupBase<Derived>::dIntegrate_product_impl(
+      const Config_t & q,
+      const Tangent_t & v,
+      const JacobianIn_t & Jin,
+      JacobianOut_t & Jout,
+      bool dIntegrateOnTheLeft,
+      const ArgumentPosition arg,
+      const AssignmentOperatorType op) const
+  {
+    Index nv_ (nv());
+    JacobianMatrix_t J (nv_, nv_);
+    dIntegrate(q, v, J, arg);
+    switch (op) {
+      case SETTO:
+        if(dIntegrateOnTheLeft) Jout = J * Jin;
```

Comment:
Maybe, we should use the noalias() here. No?

Reply:
This is a bit hazardous since Jout and Jin might be the same matrix. Another option would be to allow Jout to be either and Eigen::MatrixBase or a NoAlias so that users can call `dIntegrate(... Jout.noalias());`

Reply:
Moreover, here, we are dealing with fairly small matrices so `noalias` won't make a big difference. I think we would gain more by specializing this function for SpecialEuclidean and CartesianProduct. I prefer to keep this for later.